### DISCUSSION TITLE
Run Shell Commands

### DISCUSSION THREAD

**Discussion by johnlindquist** [johnlindquist/kit#526] [Docs]
<meta path="help/terminal-app">
      
# Run Shell Commands

Script Kit bundles [zx](https://github.com/google/zx) as the global `$`

Example from their docs (make sure to `cd` to the proper dir)

```js
await $`cat package.json | grep name`

let branch = await $`git branch --show-current`
await $`dep deploy --branch=${branch}`

await Promise.all([
  $`sleep 1; echo 1`,
  $`sleep 2; echo 2`,
  $`sleep 3; echo 3`,
])

let name = "foo bar"
await $`mkdir /tmp/${name}`
```

**Comment by Antonio-Laguna** [comment]
@johnlindquist merry Christmas! 

I've realised that, somehow, I can't run NPM and CD from the app but I can on the terminal via the kit binary, is this expected?

I'm building something that helps me release packages on https://github.com/csstools/postcss-plugins. I've managed to get it working but only on the terminal. It's more than enough but happy to debug something if it helps Scriptkit move forward.

**Comment by johnlindquist** [comment]
Did you see this?

<img width="812" alt="CleanShot 2021-12-27 at 13 06 25@2x" src="https://user-images.githubusercontent.com/36073/147503860-055ade1e-1680-4ccc-8f53-b26fba8c9135.png">

**Reply by Antonio-Laguna** [reply]
Seems that I missed that! Thanks for pointing at this. Got solved

**Reply by alexrqs** [reply]
![Screen Shot 2022-07-12 at 11 39 20 AM](https://user-images.githubusercontent.com/1099589/178547015-f219fada-3c68-49bf-83ef-163df5906cc6.png)

Was trying it but didn't work, anything I'm missing here?

**Reply by johnlindquist** [reply]
@alexrqs I'm not near a computer today to debug, but this should accomplish the same thing:

~/.kit/bin/kit set-env-var PATH ${process.env.PATH}


The end goal is the ~/.kenv/.env has a 

PATH="Your $PATH from the terminal"

Does that make sense?
